tensorflow_datasets/image/mini_imagenet.py

```diff
@@ -0,0 +1,161 @@
+_NUM_CLASSES_VALIDATION = 16
+_NUM_CLASSES_TEST = 20
+
+
+class MiniImagenet(tfds.core.GeneratorBasedBuilder):
+  """mini_imagenet: a modified version of the ILSVRC-12 dataset."""
+
+  VERSION = tfds.core.Version("1.0.2")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+      builder=self,
+      description=_MINI_IMAGENET_DESCRIPTION,
+      features=tfds.features.FeaturesDict({
+        "image": tfds.features.Image(),
+        "label": tfds.features.ClassLabel(num_classes=_NUM_CLASSES),
+      }),
+      supervised_keys=("image", "label"),
+      urls=[_MINI_IMAGENET_URL_HOME],
+      citation=_MINI_IMAGENET_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Downloads the data and defines the split."""
+
+    extracted_path = dl_manager.download_and_extract(
+    _MINI_IMAGENET_URL_DOWNLOAD)
+
+    return [
+        tfds.core.SplitGenerator(
+          name=tfds.Split.TRAIN,
+          num_shards=1,
+          gen_kwargs={
+            "idx_split": 0,
+            "path_data": os.path.join(extracted_path,
+                                      "mini-imagenet-cache-train.pkl"),
+          }
+        ),
+        tfds.core.SplitGenerator(
+          name=tfds.Split.VALIDATION,
+          num_shards=1,
+          gen_kwargs={
+            "idx_split": 1,
+            "path_data": os.path.join(extracted_path,
+                                      "mini-imagenet-cache-val.pkl"),
+          }
+        ),
+        tfds.core.SplitGenerator(
+          name=tfds.Split.TEST,
+          num_shards=1,
+          gen_kwargs={
+            "idx_split": 2,
+            "path_data": os.path.join(extracted_path,
+                                      "mini-imagenet-cache-test.pkl"),
+          }
+        ),
+      ]
+
+  def _generate_examples(self, idx_split, path_data):
+    """yields examples from the dataset.
+
+    Note: Class indices are as follows:
+    train [0, 63]; val [64, 79]; test [79, 99]
+
+    Args:
+      idx_split (int): used to identify split.
+      path_data (string): path to extracted data.
+
+    Yields:
+      dict_data (dict): data dictionary of image and label.
+        keys:
+          "image_data" (np.ndarray, shape=[num_data, 84, 84, 3],
+                        dtype=np.uint8)
+          "class_dict" (dict): class name to list of image indices.
+            value: (list of int, len=600)
+    """
+
+    # offset classass indices based on split
+    if idx_split == 0:
+      idx_class_start = 0
+    elif idx_split == 1:
+      idx_class_start = _NUM_CLASSES_TRAIN
+    elif idx_split == 2:
+      idx_class_start = _NUM_CLASSES_TRAIN + _NUM_CLASSES_VALIDATION
+
+    # read data
+    try:
+      with tf.io.gfile.GFile(path_data, "rb") as f:
+        data = pickle.load(f)
+    except UnicodeDecodeError as e:
+      with tf.io.gfile.GFile(path_data, "rb") as f:
+        data = pickle.load(f, encoding="latin1")
+    except Exception as e:
+      print("Unable to load data ", path_data, ":", e)
+      raise
```

Comment:
You should give error message as an exception message, not with print.

`raise Exception('Error message')`

Reply:
I added an exception message and removed print.